### PR TITLE
feat(linter): support eslint(default-case-last)

### PR DESCRIPTION
**Pull request by PerfectPan**

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/default_case_last.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    ///     default:
+    ///         baz();
+    ///         break;
+    ///     case 2:
+    ///         qux();
+    ///         break;
+    /// }
+    /// ```
+    DefaultCaseLast,
+    correctness
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/default_case_last.rs:53]: I wonder whether this should be a `style` rule?

File: crates/oxc_linter/src/snapshots/default_case_last.snap [hunk 2]
```diff
@@ -0,0 +1,104 @@
+---
+source: crates/oxc_linter/src/tester.rs
+assertion_line: 105
+expression: default_case_last
+---
+  ⚠ eslint(default-case-last): Enforce default clauses in switch statements to be last
+   ╭─[default_case_last.tsx:1:1]
+ 1 │ switch (foo) { default: bar(); break; case 1: baz(); break; }
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/default_case_last.snap:8]: nit: span the `default` keyword only, the body is uninteresting regarding the error. You can use `Span::new(default_clause.span.start, default_clause.span.start + 7)`.

File: crates/oxc_linter/src/rules/eslint/default_case_last.rs (hunk 1 above)
Comment (PerfectPan) [reply crates/oxc_linter/src/rules/eslint/default_case_last.rs:53]: Sorry, I didn't notice it. I will check it later.

File: crates/oxc_linter/src/snapshots/default_case_last.snap (hunk 2 above)
Comment (PerfectPan) [reply crates/oxc_linter/src/snapshots/default_case_last.snap:8]: Get.

**Pull request merged**